src/pe_reports/metrics.py

```diff
@@ -12,25 +22,571 @@ def __init__(self, start_date, end_date, org_uid):
+        self.start_date = start_date
+        self.end_date = end_date
+        self.org_uid = org_uid
+        insecure_df = query_shodan(
+            org_uid,
+            start_date,
+            end_date,
+            "shodan_insecure_protocols_unverified_vulns",
+        )
+        self.insecure_df = insecure_df
+
+        vulns_df = query_shodan(org_uid, start_date, end_date, "shodan_verified_vulns")
+        vulns_df["port"] = vulns_df["port"].astype(str)
+        self.vulns_df = vulns_df
+
+        assets_df = query_shodan(org_uid, start_date, end_date, "shodan_assets")
+        self.assets_df = assets_df
+
+    def protocol_count(self):
+        """Return a count for each insecure protocol."""
+        insecure_df = self.insecure_df
+        insecure = insecure_df[insecure_df["type"] == "Insecure Protocol"]
+        insecure = insecure[
+            (insecure["protocol"] != "http") & (insecure["protocol"] != "smtp")
+        ]
+        risky_assets = insecure[["ip", "protocol"]].drop_duplicates(keep="first")
+
+        # Horizontal bar: insecure protocol count
+        pro_count = risky_assets.groupby(["protocol"], as_index=False)["protocol"].agg(
+            {"id_count": "count"}
+        )
+        return pro_count
+
+    def risky_assets(self):
+        """Return dataframe of ip and risky protocols."""
+        insecure_df = self.insecure_df
+        insecure = insecure_df[insecure_df["type"] == "Insecure Protocol"]
+        insecure = insecure[
+            (insecure["protocol"] != "http") & (insecure["protocol"] != "smtp")
+        ]
+        risky_assets = insecure[["ip", "protocol"]].drop_duplicates(keep="first")
+        return risky_assets
+
+    def risky_ports_count(self):
+        """Return total count of insecure protocols."""
+        insecure_df = self.insecure_df
+        insecure = insecure_df[insecure_df["type"] == "Insecure Protocol"]
+        insecure = insecure[
+            (insecure["protocol"] != "http") & (insecure["protocol"] != "smtp")
+        ]
+        risky_assets = insecure[["ip", "protocol"]].drop_duplicates(keep="first")
+
+        pro_count = risky_assets.groupby(["protocol"], as_index=False)["protocol"].agg(
+            {"id_count": "count"}
+        )
+
+        # Total Open Ports with Insecure protocols
+        return pro_count["id_count"].sum()
+
+    def total_verif_vulns(self):
+        """Return total count of verified vulns."""
+        vulns_df = self.vulns_df
+        verif_vulns = (
+            vulns_df[["cve", "ip", "port"]]
+            .groupby("cve")
+            .agg(lambda x: "  ".join(set(x)))
+            .reset_index()
+        )
+
+        if len(verif_vulns) > 0:
+            verif_vulns["count"] = verif_vulns["ip"].str.split("  ").str.len()
+            verifVulns = verif_vulns["count"].sum()
+
+        else:
+            verifVulns = 0
+
+        return verifVulns
+
+    def unverified_cve(self):
+        """Return top 15 unverified cves and their counts."""
```

Comment:
Use proper capitalization for acronyms in docstrings:
```suggestion
        """Return top 15 unverified CVEs and their counts."""
```